### ISSUE TITLE
CKV_AZURE_26/27 false-positive behavior for Bicep

### ISSUE THREAD

**Issue by artecoder** [bridgecrewio/checkov#7439]
### Issue Summary

- Check ID(s): `CKV_AZURE_26`, `CKV_AZURE_27`
- Framework: `bicep`
- Current behavior: reports `FAILED` for `Microsoft.Sql/servers/databases` even when `Microsoft.Sql/servers/databases/securityAlertPolicies` exists with valid properties.
- Expected behavior: pass when `state=Enabled`, `emailAddresses` is set, and `emailAccountAdmins=Enabled`.

### Reproduction Steps

1. Run:

```powershell
checkov -f .\AzureSqlDataBaseInstances.bicep --framework bicep --check CKV_AZURE_26,CKV_AZURE_27
```

2. Observe failed checks on the SQL database resource.
3. Run the strong validation script and observe:
   - ARM normalized scan passes `CKV_AZURE_26/27`.
   - Mandatory PowerShell validation also passes.

**Comment by maxamel** [comment]
@artecoder can you share the file .\AzureSqlDataBaseInstances.bicep or any other content that will assist with reproducing this issue?

**Comment by artecoder** [comment]
Sure here I have my Bicep. The problem is that sqlServerSecurityAlertPolicyDatabase is not detected as child.


```
param maxSizeBytes int = 1073741824
param skuName string = 'S1'
param DatabaseConsecutive string = '01'
param databaseName string
param environment string = 'dev'
param keyVaultKey string
param sqlServername string
param auditStorage string
param keyVaultName string
param tags object = {
  tag1: 'value1'
  tag2: 'value2'
}
@secure()
param identityPrincipalId string

var supportsZoneRedundancy = startsWith(skuName, 'P') || startsWith(skuName, 'HS')
var supportsReadScale = startsWith(skuName, 'P') || startsWith(skuName, 'BC') || startsWith(skuName, 'HS')
var includeSampleData = (skuName == 'S0' || skuName == 'S1' || skuName == 'S2') && environment != 'pdn'
var licenseTypeActive =  startsWith(skuName, 'BC_Gen5')

var skuCapacityMap = {
  BC_Gen5: 2
  S0: 10
  S1: 20
  S2: 50
  S3: 100
  S4: 200
  S6: 400
  S7: 800
  S9: 1600
  S12: 3000
  P1: 125
  P2: 250
  P4: 500
  P6: 1000
  P11: 1750
  P15: 4000
}
param identityResourceId string
var skuCapacity = contains(skuCapacityMap, skuName) ? skuCapacityMap[skuName] : (environment == 'pdn' ? 20 : 10)

resource keyVault 'Microsoft.KeyVault/vaults@2021-10-01' existing = {
  name: keyVaultName
}

resource sqlServer 'Microsoft.Sql/servers@2023-08-01-preview' existing = {
  name: sqlServername
}

resource sqlDatabase 'Microsoft.Sql/servers/databases@2023-08-01-preview' = {
  name: '${databaseName}-${environment}-db-${DatabaseConsecutive}'
  location: resourceGroup().location
  parent: sqlServer
  sku: {
    name: skuName
    tier: startsWith(skuName, 'S') ? 'Standard' : startsWith(skuName, 'P') ? 'Premium' : startsWith(skuName, 'BC') ? 'BusinessCritical' : 'GeneralPurpose'
    capacity: skuCapacity
  }
  identity: {
    type: 'UserAssigned'
    userAssignedIdentities: {
      '${identityResourceId}': {}
    }
  }
  properties: {
    catalogCollation: 'SQL_Latin1_General_CP1_CI_AS'
    collation: 'SQL_Latin1_General_CP1_CI_AS'
    maxSizeBytes: maxSizeBytes
    sampleName: includeSampleData ? 'AdventureWorksLT' : null
    zoneRedundant: supportsZoneRedundancy
    licenseType: licenseTypeActive ? 'BasePrice' : null 
    readScale: supportsReadScale ? 'Enabled' : null
    availabilityZone: 'NoPreference'
    createMode: 'Default'
    encryptionProtector: keyVaultKey
    encryptionProtectorAutoRotation: true
  }
  tags: tags
}

resource sqlServerSecurityAlertPolicyDatabase 'Microsoft.Sql/servers/databases/securityAlertPolicies@2014-04-01' = {
  name: 'default'
  parent: sqlDatabase
  location: resourceGroup().location
  properties: {
    state: 'Enabled'
    emailAccountAdmins: 'Enabled'
    disabledAlerts: null
    emailAddresses: 'xxx@yyy.com'
  }
}

resource auditingSettingsDB 'Microsoft.Sql/servers/databases/auditingSettings@2022-05-01-preview' = {
  name: 'default'
  parent: sqlDatabase
  properties: {
    state: 'Enabled'
    storageEndpoint: auditStorage
    storageAccountSubscriptionId: subscription().subscriptionId
    isManagedIdentityInUse: true
    retentionDays: 90
  }
}
```


Output of checkov -f ./AzureSqlDataBaseInstances.bicep

```

`     _               _
   ___| |__   ___  ___| | _______   __
  / __| '_ \ / _ \/ __| |/ / _ \ \ / /
 | (__| | | |  __/ (__|   < (_) \ V /
  \___|_| |_|\___|\___|_|\_\___/ \_/

By Prisma Cloud | version: 3.2.502 

bicep scan results:

Passed checks: 4, Failed checks: 5, Skipped checks: 0

Check: CKV_AZURE_131: "SecureString parameter should not have hardcoded default values"
        PASSED for resource: string.identityPrincipalId
        File: /AzureSqlDataBaseInstances.bicep:14-15
        Guide: https://docs.prismacloud.io/en/enterprise-edition/policy-reference/azure-policies/azure-general-policies/ensure-that-security-contact-emails-is-set
Check: CKV_AZURE_229: "Ensure the Azure SQL Database Namespace is zone redundant"
        PASSED for resource: Microsoft.Sql/servers/databases.sqlDatabase
        File: /AzureSqlDataBaseInstances.bicep:57-86
        Guide: https://docs.prismacloud.io/en/enterprise-edition/policy-reference/azure-policies/azure-storage-policies/bc-azure-229
Check: CKV_AZURE_23: "Ensure that 'Auditing' is set to 'On' for SQL servers"
        PASSED for resource: Microsoft.Sql/servers/databases.sqlDatabase
        File: /AzureSqlDataBaseInstances.bicep:57-86
        Guide: https://docs.prismacloud.io/en/enterprise-edition/policy-reference/azure-policies/azure-logging-policies/bc-azr-logging-2
Check: CKV_AZURE_25: "Azure SQL Server threat detection alerts are enabled for all threat types"
        PASSED for resource: Microsoft.Sql/servers/databases.sqlDatabase
        File: /AzureSqlDataBaseInstances.bicep:57-86
        Guide: https://docs.prismacloud.io/en/enterprise-edition/policy-reference/azure-policies/azure-general-policies/bc-azr-general-6
Check: CKV_AZURE_26: "Ensure that 'Send Alerts To' is enabled for MSSQL servers"
        FAILED for resource: Microsoft.Sql/servers/databases.sqlDatabase
        File: /AzureSqlDataBaseInstances.bicep:57-86
        Guide: https://docs.prismacloud.io/en/enterprise-edition/policy-reference/azure-policies/azure-general-policies/bc-azr-general-7

                57 | resource sqlDatabase 'Microsoft.Sql/servers/databases@2023-08-01-preview' = {
                58 |   name: '${databaseName}-${environment}-db-${DatabaseConsecutive}'
                59 |   location: resourceGroup().location
                60 |   parent: sqlServer
                61 |   sku: {
                62 |     name: skuName
                63 |     tier: startsWith(skuName, 'S') ? 'Standard' : startsWith(skuName, 'P') ? 'Premium' : startsWith(skuName, 'BC') ? 'BusinessCritical' : 'GeneralPurpose'
                64 |     capacity: skuCapacity
                65 |   }
                66 |   identity: {
                67 |     type: 'UserAssigned'
                68 |     userAssignedIdentities: {
                69 |       '${identityResourceId}': {}
                70 |     }
                71 |   }
                72 |   properties: {
                73 |     catalogCollation: 'SQL_Latin1_General_CP1_CI_AS'
                74 |     collation: 'SQL_Latin1_General_CP1_CI_AS'
                75 |     maxSizeBytes: maxSizeBytes
                76 |     sampleName: includeSampleData ? 'AdventureWorksLT' : null
                77 |     zoneRedundant: true
                78 |     licenseType: licenseTypeActive ? 'BasePrice' : null
                79 |     readScale: supportsReadScale ? 'Enabled' : null
                80 |     availabilityZone: 'NoPreference'
                81 |     createMode: 'Default'
                82 |     encryptionProtector: keyVaultKey
                83 |     encryptionProtectorAutoRotation: true
                84 |   }
                85 |   tags: tags
                86 | }

Check: CKV_AZURE_27: "Ensure that 'Email service and co-administrators' is 'Enabled' for MSSQL servers"
        FAILED for resource: Microsoft.Sql/servers/databases.sqlDatabase
        File: /AzureSqlDataBaseInstances.bicep:57-86
        Guide: https://docs.prismacloud.io/en/enterprise-edition/policy-reference/azure-policies/azure-general-policies/bc-azr-general-8

                57 | resource sqlDatabase 'Microsoft.Sql/servers/databases@2023-08-01-preview' = {
                58 |   name: '${databaseName}-${environment}-db-${DatabaseConsecutive}'
                59 |   location: resourceGroup().location
                60 |   parent: sqlServer
                61 |   sku: {
                62 |     name: skuName
                63 |     tier: startsWith(skuName, 'S') ? 'Standard' : startsWith(skuName, 'P') ? 'Premium' : startsWith(skuName, 'BC') ? 'BusinessCritical' : 'GeneralPurpose'
                64 |     capacity: skuCapacity
                65 |   }
                66 |   identity: {
                67 |     type: 'UserAssigned'
                68 |     userAssignedIdentities: {
                69 |       '${identityResourceId}': {}
                70 |     }
                71 |   }
                72 |   properties: {
                73 |     catalogCollation: 'SQL_Latin1_General_CP1_CI_AS'
                74 |     collation: 'SQL_Latin1_General_CP1_CI_AS'
                75 |     maxSizeBytes: maxSizeBytes
                76 |     sampleName: includeSampleData ? 'AdventureWorksLT' : null
                77 |     zoneRedundant: true
                78 |     licenseType: licenseTypeActive ? 'BasePrice' : null
                79 |     readScale: supportsReadScale ? 'Enabled' : null
                80 |     availabilityZone: 'NoPreference'
                81 |     createMode: 'Default'
                82 |     encryptionProtector: keyVaultKey
                83 |     encryptionProtectorAutoRotation: true
                84 |   }
                85 |   tags: tags
                86 | }
```

**Comment by Eljees** [comment]
I looked into why this fails, in case it saves someone time — I'm not opening a PR because the fix looks like a design decision rather than a patch.

`CKV_AZURE_26` (`checkov/arm/checks/resource/SQLServerEmailAlertsEnabled.py`) is a `BaseResourceCheck` over `Microsoft.Sql/servers/databases`, and it decides purely from that one resource's conf:

```python
resources = conf.get("resources") or []
for idx, resource in enumerate(force_list(resources)):
    if resource.get("type") in (
        "Microsoft.Sql/servers/databases/securityAlertPolicies",
        "securityAlertPolicies",
    ):
        ...
return CheckResult.FAILED
```

So it only passes when the alert policy is **nested inside** the database resource. In the reported template the policy is a separate top-level resource attached with `parent:`, which never lands in the database's `resources` array — hence the FAILED, even though the policy exists and is valid. `CKV_AZURE_27` (`SQLServerEmailAlertsToAdminsEnabled`) has the same shape.

That also explains the asymmetry in the report: the ARM-normalized scan passes because the normalization nests the child under its parent, while the Bicep path keeps it as a sibling.

A `scan_resource_conf` check can't see sibling resources by design, so fixing this properly means either resolving `parent:` relationships into nested `resources` during Bicep→ARM conversion, or reimplementing these two as graph checks that can follow the edge from the policy to its parent database. Both are choices about where the correction belongs, which is why I'd rather flag it than pick one.

Two things I could not verify and would want confirmed before anyone acts on the above: the exact shape the Bicep parser produces for a `parent:`-attached child resource, and whether any existing Azure check already resolves that relationship in a way these two could reuse.
